fbw-a380x/src/systems/systems-host/Misc/LegacyGpws.ts

```diff
@@ -226,20 +248,41 @@ export class LegacyGpws {
   gpws(deltaTime) {
     // EGPWS receives ADR1 only
     const baroAlt = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_1_BARO_CORRECTED_ALTITUDE_1');
+    const computedAirspeed = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_1_COMPUTED_AIRSPEED');
+    const pitch = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_IR_1_PITCH');
+    const inertialVs = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_IR_1_VERTICAL_SPEED');
+    const barometricVs = Arinc429Word.fromSimVarValue('L:A32NX_ADIRS_ADR_1_BAROMETRIC_VERTICAL_SPEED');
     const radioAlt1 = Arinc429Word.fromSimVarValue('L:A32NX_RA_1_RADIO_ALTITUDE');
     const radioAlt2 = Arinc429Word.fromSimVarValue('L:A32NX_RA_2_RADIO_ALTITUDE');
     const radioAlt = radioAlt1.isFailureWarning() || radioAlt1.isNoComputedData() ? radioAlt2 : radioAlt1;
     const radioAltValid = radioAlt.isNormalOperation();
-    const onGround = SimVar.GetSimVarValue('SIM ON GROUND', 'Bool');
+    const isOnGround = !this.isAirVsGroundMode;
+
+    const isGpwsSysOff = SimVar.GetSimVarValue('L:A32NX_GPWS_SYS_OFF', 'Bool') === 1;
+    const isTerrModeOff = SimVar.GetSimVarValue('L:A32NX_GPWS_TERR_OFF', 'Bool') === 1;
+    const isFlapModeOff = SimVar.GetSimVarValue('L:A32NX_GPWS_FLAP_OFF', 'Bool') === 1;
```

Comment:
It seems that this variable is called `A32NX_GPWS_FLAPS_OFF` on the A380 whereas it's `A32NX_GPWS_FLAP_OFF` on the A320`